src/TriggerBinding/SqlScalerProvider.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    internal class SqlScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly SqlTriggerScaleMonitor _scaleMonitor;
+        private readonly SqlTriggerTargetScaler _targetScaler;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SqlScalerProvider"/> class.
+        /// Use for obtaining scale metrics by scale controller.
+        /// </summary>
+        /// <param name="serviceProvider"></param>
+        /// <param name="triggerMetadata"></param>
+        public SqlScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            IConfiguration config = serviceProvider.GetService<IConfiguration>();
+            ILoggerFactory loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            ILogger logger = loggerFactory.CreateLogger(LogCategories.CreateTriggerCategory("Sql"));
+            SqlMetaData sqlMetadata = JsonConvert.DeserializeObject<SqlMetaData>(triggerMetadata.Metadata.ToString());
+            sqlMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
+            var userTable = new SqlObject(sqlMetadata.TableName);
+            string connectionString = SqlBindingUtilities.GetConnectionString(sqlMetadata.ConnectionStringSetting, config);
+            int maxChangesPerWorker = config.GetValue(SqlTriggerConstants.ConfigKey_SqlTrigger_MaxChangesPerWorker, SqlOptions.DefaultMaxChangesPerWorker);
```

Comment:
You need to be reading from the SqlOptions we added a little while back too - e.g. https://github.com/Azure/azure-functions-sql-extension/blob/main/src/TriggerBinding/SqlTriggerListener.cs#L79

PR: https://github.com/Azure/azure-functions-sql-extension/pull/960

Reply:
Changed to read from this section in config `$"azureWebJobs:extensions:sql:MaxChangesPerWorker"`. I believe this is the path for setting SqlOptions?